infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientServerRouter.java

```diff
@@ -535,7 +526,13 @@ public void receive(CorfuMessage.ResponseMsg msg) {
                 completeRequest(msg.getHeader().getSession(), msg.getHeader().getRequestId(), null);
                 return;
 
-            } else {
+            } else if (msg.getPayload().getPayloadCase().equals(CorfuMessage.ResponsePayloadMsg.PayloadCase.LR_SINK_SESSION_INITIALIZATION_ACK)){
+                log.debug("Received an ACK for sink side session initialization {}/{}", msg.getHeader().getRequestId(),
+                        msg.getPayload().getPayloadCase());
+                completeRequest(msg.getHeader().getSession(), msg.getHeader().getRequestId(), null);
```

Comment:
This makes sense, so we can remove the previously pointed `handleSinkSessionCreationResponse` in LogReplicationServer